### PR TITLE
helmfile 0.21.0 (new formula)

### PR DESCRIPTION
**Pull request by wendorf**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/helmfile.rb
Comment (fxcoudert) [inline Formula/helmfile.rb]: We need a test that exercises the some of the functionality of the app.

**Review by fxcoudert — DISMISSED** [review]

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Helmfile is a coordinating wrapper around Helm, so I've added helm as a dependency, and in the tests initialized it and used it. I'm unclear on best practices for declaring this sort of dependency. Since it's runtime-only, but may be installed outside of Homebrew, is it still appropriate to list as a dependency? Or should it just be a test dependency?

My update to this PR treats it as a normal dependency, but please let me know if this needs more adjusting.

Comment (fxcoudert) [reply Formula/helmfile.rb]: `depends_on "kubernetes-helm" => :recommended` seems like the right way to go

Comment (wendorf) [reply Formula/helmfile.rb]: It looks like the best practices for new formulae just changed 9 days ago (Homebrew/brew#4170) and there are now errors for both `:recommended` and `:optional` dependencies:

```
→ brew audit --new-formula helmfile
helmfile:
  * Formulae should not have [:recommended] dependencies.
Error: 1 problem in 1 formula
```

```
→ brew audit --new-formula helmfile
helmfile:
  * Formulae should not have [:optional] dependencies.
Error: 1 problem in 1 formula
```

Should this just be merged as-is? Should a follow-up PR add the option?

File: Formula/helmfile.rb
Comment (DomT4) [inline Formula/helmfile.rb]: You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.

File: Formula/helmfile.rb
Comment (DomT4) [inline Formula/helmfile.rb]: This should be wrapped to 80.

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Fixed. Thanks!

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Fixed. Thanks!

**Comment by DomT4** [comment]
@ilovezfs Looks like CI needs to do this one again; it blew up 🙃.

**Comment by ilovezfs** [comment]
🎡

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Realized I missed the 80 column limit in the other comment as well. Fixed them both.

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Switched

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Changed

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Removed

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Removed

**Comment by estahn** [comment]
Where is this at?

**Comment by wendorf** [comment]
I believe I've addressed all the feedback, so the PR should be ready for re-review.

File: Formula/helmfile.rb
Comment (DomT4) [inline Formula/helmfile.rb]: The regex use without actually checking a regex is kind of upsetting, but I'm not sure:
```ruby
output = "\"stable\" has been added to your repositories"
```
is especially stylistically pleasing either 😭.

File: Formula/helmfile.rb
Comment (DomT4) [inline Formula/helmfile.rb]: There's a version update available now. The last CI build has been lost to churn so might as well push the update rather than asking for another rebuild manually.

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: I swapped to single quotes expecting to see a style error, but this passed:

```
brew test helmfile
brew audit --new-formula helmfile
brew audit --strict --online helmfile
brew style helmfile
```

Is there an audit/style check I missed, or is single-quotes good? I'm assuming it's good, so have pushed that change.

File: Formula/helmfile.rb
Comment (wendorf) [reply Formula/helmfile.rb]: Just bumped to 0.19.0

File: Formula/helmfile.rb
Comment (DomT4) [inline Formula/helmfile.rb]: `system bin/"helm", "init", "--client-only"`

Comment (DomT4) [reply Formula/helmfile.rb]: Oh, unless `helm` is part of `kubernetes-helm`?

Comment (wendorf) [reply Formula/helmfile.rb]: Correct, `helm` is the main binary `kubernetes-helm` installs.

Comment (DomT4) [reply Formula/helmfile.rb]: Eh. Gonna throw this one to @ilovezfs for a ruling then. Personally I prefer when we reference other formulae we do things like this:

```ruby
system Formula["kubernetes-helm"].opt_bin/"helm", "init", "--client-only"
```
but since Joe runs this place these days I'll let him decide on how fussy he wants to be.

Comment (wendorf) [reply Formula/helmfile.rb]: I certainly don't mind being fussy. Just pushed the change.

Comment (DomT4) [reply Formula/helmfile.rb]: I don't think it's an absolutely mandatory thing in general; I just prefer the longhand because it's clearer & slightly more resilient. Thanks for the change ❤️

**Review by DomT4 — APPROVED** [review]
It's good with me but will need Joe/FX/etc to look it over.

**Comment by gaieges** [comment]
Pinging on this, would love to use it

**Comment by DomT4** [comment]
There's an update available here from upstream but this needs a maintainer to look over this still.

**Comment by wendorf** [comment]
Just bumped to the latest (0.20.0). Would love to see this merged.

**Comment by darrenhaken** [comment]
Any update on this?

**Comment by DomT4** [comment]
This needs bumping again, but primarily needs @ilovezfs's eyes 😇.

**Comment by wendorf** [comment]
@DomT4 Thanks for the catch! Just bumped it again, this time to 0.21.0.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/7ae10498c3fb2ce59887d7fb43e049df3b99b814. Thank you for your contribution to Homebrew @wendorf; we appreciate it! :smiley_cat:

**Pull request closed without merging**
